.github/workflows/pr.yaml

```diff
@@ -0,0 +1,84 @@
+# Pull Request Workflow
+#
+# This workflow orchestrates quality checks, tests, and security scans for
+# pull requests using reusable workflows.
+#
+# Key Features:
+# - Code quality validation
+# - Test suite execution
+# - Security scanning
+# - Concurrent execution handling
+# - Automated feedback
+#
+# Process Stages:
+# 1. Quality Checks:
+#    - Code style verification
+#    - Type checking
+#    - Pre-commit hook validation
+#
+# 2. Testing:
+#    - Unit test execution
+#    - Integration testing
+#    - Coverage reporting
+#
+# 3. Security:
+#    - Changed files scanning
+#    - Vulnerability detection
+#    - Security report generation
+#
+# Required Secrets:
+# - CODECOV_TOKEN: Coverage reporting token
+#
+# Example Usage:
+# Automatically triggered on:
+# 1. Pull requests to main branch
+# 2. Pull requests to feature/* branches
+#
+# Note: Configured to cancel outdated runs when new commits are pushed
+
+name: PR Checks
+
+on:
+  pull_request:
+    branches: [main, "feature/**"]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Code quality job using reusable workflow
+  quality:
+    uses: ./.github/workflows/_reusable-code-quality.yaml
+    with:
+      python-version: "3.10"
+
+  # Test suite job using reusable workflow
+  unit-tests:
+    uses: ./.github/workflows/_reusable-test-suite.yaml
+    with:
+      test-type: "unit"
+      runner: "ubuntu-latest"
+      timeout: 10
+    secrets:
+      codecov-token: ${{ secrets.CODECOV_TOKEN }}
+
+  integration-tests:
+    uses: ./.github/workflows/_reusable-test-suite.yaml
+    with:
+      test-type: "integration"
+      runner: "self-hosted"
```

Comment:
We might have to revisit this as the org runners are also tagged `self-hosted` and they don't have a CUDA supported GPU. That's why we use `anomalib` tag for our self-hosted GPU

Reply:
if we have a tag argument here would it solve it?

Reply:
I just checked, the org runners do not seem to have self-hosted tags. So maybe it will work

Reply:
alright, I don't expect this pipeline to work out of the box, so we might need to polish it as we go